pkg/crypto/helpers_test.go

```diff
@@ -0,0 +1,35 @@
+package crypto
+
+import (
+	"testing"
+)
+
+func TestTopicFromValidCert(t *testing.T) {
+	certificate, _ := ReadPEMCertificateFile("testdata/mock_push_cert.pem")
+	pushTopic, err := TopicFromCert(certificate)
+
+	if err != nil {
+		t.Errorf("fail %s", err)
+	}
+
+	if have, want := pushTopic, "com.apple.mgmt.External.18a16429-886b-41f1-9c30-2bd04ae4fc37"; have != want {
+		t.Errorf("have %d, want %d", have, want)
+	}
+}
+
+func TestTopicFromCertWithWrongUidPrefix(t *testing.T) {
```

Comment:
IMO these tests are a bit too short. A common go pattern is to write a [table test](https://github.com/golang/go/wiki/TableDrivenTests).

[Subtests](https://blog.golang.org/subtests) also help a lot when working with a table test. 

And here's an example in the wild. https://github.com/oklog/oklog/blob/7fa04d3c21f8fedb54940c069c02e450c7cc8eb3/pkg/store/overlap_test.go

Reply:
I'll check out the table test patterns. I agree that a whole function for "one liner tests" is not ideal.